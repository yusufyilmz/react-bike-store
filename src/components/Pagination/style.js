

import styled from 'styled-components';


export const CenteredDiv = styled.div`
    text-align: center;
`


export const Div = styled.div`
    display: inline-block;
    text-align: center;
`

export const A = styled.a`
    color: black;
    float: left;
    padding: 8px 16px;
    text-decoration: none;
    transition: background-color .3s;
    border: 1px solid #ddd;
`
A.displayName = 'a';

export const ActiveA = styled(A)`
    background-color: #1abc9c;
    color: white;
    border: 1px solid #4CAF50;
`;