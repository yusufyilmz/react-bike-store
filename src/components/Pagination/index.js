

import React, { Component } from 'react';
import { Div, A, ActiveA, CenteredDiv } from './style';
import uniqueid from 'lodash.uniqueid';

class Pagination extends Component {

    getPages() {
        var pages = []
        for (let i = 0; i < this.props.pageCount; i++) {
            const MyA = i === this.props.selectedPage - 1 ? ActiveA : A;
            pages.push(<MyA key={uniqueid()} onClick={() => this.props.selectPage(i + 1)}>{i + 1}</MyA>)
        }
        return pages;
    }

    render() {
        return (
            <CenteredDiv>
                <Div>
                    <A key={uniqueid()} onClick={this.props.onPreviousPage} >&laquo;</A>
                    {this.getPages()}
                    <A key={uniqueid()}  >...</A>
                    <A key={uniqueid()} onClick={this.props.onNextPage} >&raquo;</A>
                </Div>
            </CenteredDiv>

        );
    }

}

export default Pagination;



