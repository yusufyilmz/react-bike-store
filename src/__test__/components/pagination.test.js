

import React from 'react';
import Pagination from '../../components/Pagination';
import { shallow } from 'enzyme'


describe('Pagination component', () => {

    let wrapper, pageCount;

    beforeEach(() => {
        pageCount = 1;
        wrapper = shallow(
            <Pagination pageCount={pageCount}  selectedPage={pageCount}/>
        );
    });

    test('renders element correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('renders onclick correctly', () => {
        expect(wrapper.find('a')).toHaveLength(3) 
    });
})
