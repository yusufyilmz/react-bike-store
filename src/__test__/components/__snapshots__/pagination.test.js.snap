// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination component renders element correctly 1`] = `
<styled.div>
  <styled.div>
    <a
      key="1"
    >
      «
    </a>
    <Styled(a)
      key="2"
      onClick={[Function]}
    >
      1
    </Styled(a)>
    <a
      key="3"
    >
      ...
    </a>
    <a
      key="4"
    >
      »
    </a>
  </styled.div>
</styled.div>
`;
